Scalar can be decorated with an undeclared directive

The schema below raise an error. `name` is decorated with an undeclared directive.
```graphql
type User {
    name: String! @undeclareddirective
}
```
```
panic: graphql: directive "undeclareddirective" not found
```

But, the schema below does not raise an error. `scalar S` is decorated with an undeclared directive.
```graphql
scalar S @undeclareddirective
```

Is this a bug?
The same is true not only for scalar, but also for argument, interface, etc.
The two schemas below also do not raise errors.

```graphql
type Query {
    hello(name: String! @undeclareddirective): String!
}
```

```graphql
interface I @undeclareddirective {
    name: String!
}
```

--- a/internal/schema/schema.go
+++ b/internal/schema/schema.go
@@ -268,8 +268,18 @@ func resolveNamedType(s *types.Schema, t types.NamedType) error {
 				return err
 			}
 		}
+		if err := resolveDirectives(s, t.Directives, "INTERFACE"); err != nil {
+			return err
+		}
 	case *types.InputObject:
 		if err := resolveInputObject(s, t.Values); err != nil {
+			return err
+		}
+		if err := resolveDirectives(s, t.Directives, "INPUT_OBJECT"); err != nil {
+			return err
+		}
+	case *types.ScalarTypeDefinition:
+		if err := resolveDirectives(s, t.Directives, "SCALAR"); err != nil {
 			return err
 		}
 	}
@@ -335,6 +345,11 @@ func resolveInputObject(s *types.Schema, values types.ArgumentsDefinition) error
 			return err
 		}
 		v.Type = t
+
+		if err := resolveDirectives(s, v.Directives, "ARGUMENT_DEFINITION"); err != nil {
+			return err
+		}
+
 	}
 	return nil
 }

--- a/internal/schema/schema_test.go
+++ b/internal/schema/schema_test.go
@@ -936,6 +936,60 @@ Second line of the description.
 				return nil
 			},
 		},
+		{
+			name: "Decorating scalar with an undeclared directive should return an error",
+			sdl: `
+			scalar S @undeclareddirective
+			`,
+			validateError: func(err error) error {
+				prefix := `graphql: directive "undeclareddirective" not found`
+				if err == nil || !strings.HasPrefix(err.Error(), prefix) {
+					return fmt.Errorf("expected error starting with %q, but got %q", prefix, err)
+				}
+				return nil
+			},
+		},
+		{
+			name: "Decorating argument with an undeclared directive should return an error",
+			sdl: `
+			type Query {
+				hello(name: String! @undeclareddirective): String!
+			}
+			`,
+			validateError: func(err error) error {
+				prefix := `graphql: directive "undeclareddirective" not found`
+				if err == nil || !strings.HasPrefix(err.Error(), prefix) {
+					return fmt.Errorf("expected error starting with %q, but got %q", prefix, err)
+				}
+				return nil
+			},
+		},
+		{
+			name: "Decorating input object with an undeclared directive should return an error",
+			sdl: `
+			input InputObject @undeclareddirective{}
+			`,
+			validateError: func(err error) error {
+				prefix := `graphql: directive "undeclareddirective" not found`
+				if err == nil || !strings.HasPrefix(err.Error(), prefix) {
+					return fmt.Errorf("expected error starting with %q, but got %q", prefix, err)
+				}
+				return nil
+			},
+		},
+		{
+			name: "Decorating interface with an undeclared directive should return an error",
+			sdl: `
+			interface I @undeclareddirective {}
+			`,
+			validateError: func(err error) error {
+				prefix := `graphql: directive "undeclareddirective" not found`
+				if err == nil || !strings.HasPrefix(err.Error(), prefix) {
+					return fmt.Errorf("expected error starting with %q, but got %q", prefix, err)
+				}
+				return nil
+			},
+		},
 	} {
 		t.Run(test.name, func(t *testing.T) {
 			s, err := schema.ParseSchema(test.sdl, test.useStringDescriptions)

